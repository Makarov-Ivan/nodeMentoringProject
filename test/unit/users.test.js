/* eslint-disable */
const userValidationService = require('../../src/user/services/validation');

describe('users', () => {
  describe('validation service', () => {
    describe('user object', () => {
      it('should return error, if invalid object is passed', async () => {
        const invaliUser = {
          log: 'login',
          index: 10,
          pw: 'invalid',
          age: 0,
        };
        const validationResult = await userValidationService.user(invaliUser);
        expect(validationResult).toMatchSnapshot();
      });
      it('should return undefined, if valid object is passed', async () => {
        const validUser = {
          id: 989,
          login: 'Bormasuk10',
          password: 'vAlq2qwqwe',
          age: 10,
          isDeleted: true,
        };
        const validationResult = await userValidationService.user(validUser);
        expect(validationResult).toBe(undefined);
      });
    });
    describe('querry string', () => {
      it('should return error, if invalid object is passed', async () => {
        const invaliQuerry1 = {
          loginSubstring: 'abd',
        };
        const invaliQuerry2 = {
          limit: 10,
        };
        const invaliQuerry3 = {};
        const validationResult1 = await userValidationService.querry(invaliQuerry1);
        expect(validationResult1).toMatchSnapshot();
        const validationResult2 = await userValidationService.querry(invaliQuerry2);
        expect(validationResult2).toMatchSnapshot();
        const validationResult3 = await userValidationService.querry(invaliQuerry3);
        expect(validationResult3).toMatchSnapshot();
      });
      it('should return undefined, if valid object is passed', async () => {
        const validQuerry = {
          loginSubstring: 'abd',
          limit: 10,
        };
        const validationResult = await userValidationService.querry(validQuerry);
        expect(validationResult).toBe(undefined);
      });
    });
  });
});
