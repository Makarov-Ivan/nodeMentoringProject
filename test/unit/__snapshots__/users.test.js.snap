// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`users validation service querry string should return error, if invalid object is passed 1`] = `
Array [
  "\\"limit\\" is required",
]
`;

exports[`users validation service querry string should return error, if invalid object is passed 2`] = `
Array [
  "\\"loginSubstring\\" is required",
]
`;

exports[`users validation service querry string should return error, if invalid object is passed 3`] = `
Array [
  "\\"loginSubstring\\" is required",
  "\\"limit\\" is required",
]
`;

exports[`users validation service user object should return error, if invalid object is passed 1`] = `
Array [
  "\\"login\\" is required",
  "\\"id\\" is required",
  "\\"age\\" must be greater than or equal to 4",
  "\\"log\\" is not allowed",
  "\\"index\\" is not allowed",
  "\\"pw\\" is not allowed",
]
`;
